qutip/core/data/dense.pyx

```diff
@@ -121,7 +126,9 @@ cdef class Dense(base.Data):
         low-level C code).
         """
         cdef Dense out = Dense.__new__(Dense)
-        cdef size_t size = self.shape[0]*self.shape[1]*sizeof(double complex)
+        cdef size_t size = (
+            _mul_checked(self.shape[0], self.shape[1]) * sizeof(double complex)
```

Comment:
Can the multiplication by `sizeof(double complex)` also overflow?

Reply:
`size_t` is can be `uint32` on older system so I added the check there also.